### PR TITLE
Increase transparency of red ice in CTMC

### PR DESCRIPTION
**Pull request by Hexocyte**
When "Chest Appearance Matches Contents" is enabled, the transparency of red ice, except red ice walls, is increased to make it easier to see the chest textures.

Before:
screenshot

After:
screenshot

Tested in Dolphin 5.0-16101 and RetroArch/Mupen64Plus-Next.

### CODE REVIEW

**Comment by Hexocyte** [comment]
I'm figuring out how to add ASM hacks, so I'm redoing this so that it only affects the red ice crystals with chests.

**Comment by Hexocyte** [comment]
I made it specific to the red ice crystals with chests and lowered the opacity slightly further. I updated the PR title and description above.

**Comment by cjohnson57** [comment]
Hi! Unfortunately this PR is a bit out of date now, if you update it to latest I'll probably approve it.

**Comment by r0bd0g** [comment]
Now that we're changing the chest textures again, is this change still necessary?

**Comment by Hexocyte** [comment]
If I updated this and still put all the CTMC code in Patches.py into an if block as I originally did, it'd conflict with #1559. I think the big if block makes it neater, so I'll wait until that PR is merged before I change this.

In #1502 chests are drawn in front of red ice and flame circles (even when Invisible Chests is off). That might be a better way to improve visibility? Maybe not, since it can look like the chest really is in front of the ice.

File: ASM/src/hacks.asm
Comment (mracsys) [inline ASM/src/hacks.asm:1247]: What is the next instruction in the delay slot? Is that affected by any of the new code?

This should also be jal, not j.

File: Patches.py
Comment (mracsys) [inline Patches.py:1746]: No need for putting the CSMC/CTMC changes in the same if block as the red ice changes in my opinion. Also makes it easier to merge with other open PRs if separated.

File: Patches.py
Comment (mracsys) [inline Patches.py:2183]: In my opinion, transparency changes should be generic for all red ice blocks, not just for frozen chests.

**Review by mracsys — COMMENTED** [review]
Has this been tested on console?

File: Patches.py
Comment (cjohnson57) [reply Patches.py:2183]: I agree with this.

File: ASM/src/hacks.asm
Comment (Hexocyte) [reply ASM/src/hacks.asm]: The instruction in the delay slot sets the action function and is unaffected. I added a comment with the instruction.

I used j instead of jal as a shortcut to avoid having to push the old ra to the stack. I don't think there's enough room in the function to do that properly. red_ice_alpha does the skipped instructions instead.

File: Patches.py
Comment (Hexocyte) [reply Patches.py]: I removed the Python changes when changing it to also affect red ice without chests.

File: Patches.py
Comment (Hexocyte) [reply Patches.py]: Done. It now affects all red ice except red ice walls.

**Comment by Hexocyte** [comment]
Now I made it so that it applies to all red ice except red ice walls. I also added more comments and tweaked the opacity again.

**Comment by Hexocyte** [comment]
> Now that we're changing the chest textures again, is this change still necessary?

I think it still helps. The front of the major item chest didn't change much, and I figure even with improved textures, it's bound to be at least a little hard sometimes to distinguish the chest types in full-opacity red ice.

**Review by cjohnson57 — APPROVED** [review]

**Pull request merged**
